host/rpi/rpi.go

```diff
@@ -124,28 +120,65 @@ func (d *driver) Init() (bool, error) {
 		return false, errors.New("Raspberry Pi board not detected")
 	}
 
-	// Initialize Version.
+	// Setup headers based on board revision.
 	//
 	// This code is not futureproof, it will error out on a Raspberry Pi 4
 	// whenever it comes out.
+	// Revision codes from: http://elinux.org/RPi_HardwareHistory
+	has26PinP1Header := false
+	has40PinP1Header := false
+	hasP5Header := false
+	hasAudio := false
+	hasNewAudio := false
+	hasHDMI := false
 	rev := distro.CPUInfo()["Revision"]
 	if i, err := strconv.ParseInt(rev, 16, 32); err == nil {
 		// Ignore the overclock bit.
 		i &= 0xFFFFFF
-		if i < 0x20 {
-			Version = 1
-		} else if i == 0xa01041 || i == 0xa21041 {
-			Version = 2
-		} else if i == 0xa02082 || i == 0xa22082 {
-			Version = 3
-		} else {
+		switch i {
+		case 0x0002, 0x0003: // B v1.0
+			has26PinP1Header = true
+			hasAudio = true
+		case 0x0004, 0x0005, 0x0006, // B v2.0
+			0x0007, 0x0008, 0x0009, // A v2.0
+			0x000d, 0x000e, 0x000f: // B v2.0
+			has26PinP1Header = true
+			// Only the v2 PCB has the P5 header.
+			hasP5Header = true
+			hasAudio = true
+			hasHDMI = true
+		case 0x0010, // B+ v1.0
+			0x0012,             // A+ v1.1
+			0x0013,             // B+ v1.2
+			0x0015,             // A+ v1.1
+			0x90021,            // A+ v1.1
+			0x90032,            // B+ v1.2
+			0xa01040,           // 2 Model B v1.0
+			0xa01041, 0xa21041, // 2 Model B v1.1
+			0xa22042, // 2 Model B v1.2
+			0x900092, // Zero v1.2
+			0x900093, // Zero v1.3
+			0x920093, // Zero v1.3
+			0x9000c1: // Zero W v1.1
+			has40PinP1Header = true
+			hasAudio = true
```

Comment:
I don't consider the Pi Zero to have "proper audio" so I'd prefer to not register these.